rest/http_client.go

```diff
@@ -102,11 +128,17 @@ func (c *HTTPClientImpl) DoWithHeaders(ctx context.Context, compiledRoute *route
 		return NewError(rs, err)
 	}
 
+	// catch rate limit errors here
+	// TODO: we should retry on hitting 429
```

Comment:
unless the API can incorrectly 429, no we should just wait for retry-after

Reply:
Yup 
This was meant with the correct delay
Just need to grab the bucket again in a loop over retry count